packages/runtime/datastore/src/channelDeltaConnection.ts

```diff
@@ -116,7 +117,7 @@ export class ChannelDeltaConnection implements IDeltaConnection {
 
 	public applyStashedOp(content: any): unknown {
 		try {
-			this.stashedOpMd = createStashedOpMetadata();
+			this.stashedOpMd = this.isAttachedAndVisible() ? createStashedOpMetadata() : undefined;
```

Comment:
only create and return stashed op metadata if we are attached and visible, otherwise we'll just pass undefined up, and it will be dropped by pending state manager